### PR TITLE
Showing Restaurants and Header using React & Bootstrap

### PR DESCRIPTION
**Pull request by maximveksler**
Rona & Yon - Please share some feedback guys

### CODE REVIEW

**Comment by maximveksler** [comment]
@RonaTross @yonbergman please feedback.

File: index.html [hunk 1]
```diff
@@ -0,0 +1,16 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+  <meta charset="utf-8">
+  <title>Setup</title>
+  <!-- <link rel="stylesheet" href="style.css"> -->
```
Comment (RonaTross) [inline index.html]: remove commented code

File: Header.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import React from 'react'
+
+var captionCSS = {
+  position: 'absolute'
+  ,top: '45%'
+  ,left: 0
+  ,width: '100%'
+}
+
+var backgroundImageCSS = {
```
Comment (RonaTross) [inline Header.js:3]: should be in css file

File: Header.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import React from 'react'
+
+var captionCSS = {
```
Comment (RonaTross) [inline Header.js]: should be in css file

File: Restaurant.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+    var foodTypeImage = "";
+    if (this.model.foodType === "Burger") {
+      foodTypeImage = "assets/img/Burger.png"
+    } else {
+      foodTypeImage = "http://placeponi.es/48/48"
+    }
+
+    this.state = {
+      foodTypeImage: foodTypeImage
```
Comment (RonaTross) [inline Restaurant.js]: the image would fit better to props, since it doesn't change.

File: Restaurant.js [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    }
+  }
+
+  render() {
+    var ratings = [];
+    for (var i=0; i < this.model.rating; i++) {
```
Comment (RonaTross) [inline Restaurant.js]: using .map is better

File: RestaurantModel.js [hunk 6]
```diff
@@ -0,0 +1,9 @@
+class RestaurantModel {
+  constructor(name, foodType, rating) {
```
Comment (RonaTross) [inline RestaurantModel.js]: i'd add doc about the parameter types

File: Restaurant.js [hunk 7]
```diff
@@ -0,0 +1,55 @@
+import React from 'react'
+import RestaurantModel from './RestaurantModel'
+
+class Restaurant extends React.Component {
+  constructor(props) {
+    super(props)
+    this.displayName = "Restaurant"
+
+    this.model = props.model
```
Comment (RonaTross) [inline Restaurant.js]: I haven't seen this pattern before, and I'm afraid it could cause bugs in some edge cases

File: Restaurant.js [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+
+    var foodTypeImage = "";
+    if (this.model.foodType === "Burger") {
```
Comment (RonaTross) [inline Restaurant.js]: this would fit better inside render (and then it also won't be needed in the state)

File: index.html (hunk 1 above)
Comment (maximveksler) [inline index.html]: fixed by https://github.com/maximveksler/EatWell/pull/1/commits/fca7e31d98134070fc08914d10c175e337d04c01

File: Header.js (hunk 2 above)
Comment (maximveksler) [inline Header.js:3]: fixed by https://github.com/maximveksler/EatWell/pull/1/commits/fca7e31d98134070fc08914d10c175e337d04c01

File: Header.js (hunk 3 above)
Comment (maximveksler) [inline Header.js]: fixed by https://github.com/maximveksler/EatWell/pull/1/commits/fca7e31d98134070fc08914d10c175e337d04c01

File: Restaurant.js (hunk 8 above)
Comment (maximveksler) [inline Restaurant.js]: fixed by https://github.com/maximveksler/EatWell/pull/1/commits/cee837cd15db5a5bf84289c75fc191d6297c362c

File: Restaurant.js (hunk 7 above)
Comment (maximveksler) [inline Restaurant.js]: fixed by https://github.com/maximveksler/EatWell/pull/1/commits/cee837cd15db5a5bf84289c75fc191d6297c362c

File: Restaurant.js (hunk 5 above)
Comment (maximveksler) [inline Restaurant.js]: @RonaTross I agree. Sadly it's an integer, can't find a good alternative besides using underscore.js but it doesn't seem worth it. Any syntax alternatives in the form of iterators / other js stuff? I'd be glad to learn.

Comment (maximveksler) [inline Restaurant.js]: Also check this out http://codereview.stackexchange.com/questions/126616/better-way-to-iterate-over-integer-in-javascript (who's the hero of the day? me.)

**Comment by maximveksler** [comment]
So i'm merging this into master, please add more comments as you go. I will be fixing and merging further as we go along.

File: Restaurant.js (hunk 5 above)
Comment (RonaTross) [inline Restaurant.js]: Sorry, misunderstood it :) thought rating was also an array :)

Comment (maximveksler) [inline Restaurant.js]: good. this is the last one then. I'm merging.

**Comment by yonbergman** [comment]
Please include screenshot or gif to see how it looks

@maxim: adding.

screen shot 2016-04-25 at 22 26 01

File: Header.js [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+var backgroundImageCSS = {
+}
+
+class Header extends React.Component {
+  render() {
+    return (
+      <div className="jumbotron text-center eat-header-caption">
+        <div className="container">
+          <div className="row" >
+            <div className="col-md-12" >
```
Comment (yonbergman) [inline Header.js:12]: Just a bootstrap thing but if you have a col md 12 means that when its on a medium sized screen it should be a full row. That's also the default behavior. So it's kind of redundant

File: Restaurant.js [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+
+class Restaurant extends React.Component {
+  constructor(props) {
+    super(props)
+    this.displayName = "Restaurant"
+  }
+
+  render() {
+    // helper for code simplicity.
+    var model = this.props.model
```
Comment (yonbergman) [inline Restaurant.js:12]: I wouldn't wrap the components data under model of possible

File: Restaurant.js [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+
+    // Controller calculats food icon for HTML View presentation.
+    var foodTypeImage = "";
+    if (model.foodType === "Burger") {
+      foodTypeImage = "assets/img/Burger.png"
+    } else {
+      foodTypeImage = "http://placeponi.es/48/48"
+    }
+
```
Comment (yonbergman) [inline Restaurant.js:21]: Extract this to its own method perhaps

File: Restaurant.js [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+    // Prepare rating JSX
+    var ratings = [];
+    for (var i=0; i < model.rating; i++) {
+        ratings.push(<span className="glyphicon glyphicon-star" key={i}/>);
+    }
```
Comment (yonbergman) [inline Restaurant.js:26]: Include underscore or lodash, and use the times method. It's much nicer b

File: RestaurantList.js [hunk 13]
```diff
@@ -0,0 +1,35 @@
...
+  }
+
+  render() {
+    return (
+      <div className="row">
+        <div className="col-md-7">
+          <ul class="media-list">
+            {this.state.restaurants.map(function(restaurant, i) {
+              return <Restaurant model={restaurant} key={i} />
+            })}
```
Comment (yonbergman) [inline RestaurantList.js:27]: I would try to not run maps inside html tags. It makes it harder to read the Dom structure.

**Comment by yonbergman** [comment]
👍👾

File: Header.js (hunk 9 above)
Comment (maximveksler) [inline Header.js:12]: fixed by https://github.com/maximveksler/EatWell/commit/18ab3068094d021cb7807a586a495e8a5ba8492a

File: Restaurant.js (hunk 10 above)
Comment (maximveksler) [inline Restaurant.js:12]: My direction is that `ResaurantModel` is a JSON coming from REST end point, I'd assume it won't be a React object. Open for discussion..

File: Restaurant.js (hunk 11 above)
Comment (maximveksler) [inline Restaurant.js:21]: fixed by https://github.com/maximveksler/EatWell/commit/e3e9e11d501c708c526bf5e3ba3b47d15b18795d

File: Restaurant.js (hunk 12 above)
Comment (maximveksler) [inline Restaurant.js:26]: lodash is awesome, it has a babel plugin that allows me to `import lodash` and have the plugin take only the chunks of the library that I'm actually using. I hope all 3rd party code in the world would behave in a similar way.

This is fixed by https://github.com/maximveksler/EatWell/commit/876050d03e9f54109467834e7fcffeaca17a680e and https://github.com/maximveksler/EatWell/commit/120cb1ae48afe9fd28ca5fa0db46034c43e40d2d

File: RestaurantList.js (hunk 13 above)
Comment (maximveksler) [inline RestaurantList.js:27]: I agree, your right. Does looks nicer.

Fixed by https://github.com/maximveksler/EatWell/commit/b79f59dd6b677b409f7a410365517417a13fc5fd

**Pull request merged**